source/CIS/cis-2.rst

```diff
@@ -0,0 +1,790 @@
+    - A URL.
+  * - ``hash`` (optional)
+    - string
+    - A SHA256 hash of the URL content encoded as a hex string.
+
+Attributes are objects with the following fields:
+
+.. list-table:: Attribute JSON object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``type``
+    - string
+    - Type for the value field of the attribute.
+  * - ``name``
+    - string
+    - Name of the attribute.
+  * - ``value``
+    - string
+    - Value of the attrbute.
+
+
+Example token metadata: Fungible
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An example of token metadata for a CIS2 implementation wrapping the CCD could be:
+
+.. code-block:: json
+
+  {
+    "name": "Wrapped CCD Token",
+    "symbol": "wCCD",
+    "decimals": 6,
+    "description": "A CIS2 token wrapping the Concordium native token (CCD)",
+    "thumbnail": { "url": "https://location.of/the/thumbnail.png" },
+    "display": { "url": "https://location.of/the/display.png" },
+    "artifact": { "url": "https://location.of/the/artifact.png" },
+    "localization": {
+      "da-DK": {
+        "url": "https://location.of/the/danish/metadata.json",
+        "hash": "624a1a7e51f7a87effbf8261426cb7d436cf597be327ebbf113e62cb7814a34b"
+      }
+    }
+  }
+
+The danish localization JSON file could be:
+
+.. code-block:: json
+
+  {
+    "description": "CIS2 indpakket CCD"
+  }
+
+Example token metadata: Non-fungible
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An example of token metadata for a NFT could be:
+
+.. code-block:: json
+
+  {
+    "name": "Bibi - The Ryan Cat",
+    "description": "Ryan cats are lonely creatures travelling the galaxy in search of their ancestors and true inheritance",
+    "thumbnail": { "url": "https://location.of/the/thumbnail.png" },
+    "display": { "url": "https://location.of/the/display.png" },
+    "attributes": [{
+      "type": "date",
+      "name": "Birthday",
+      "value": "1629792199610"
+    }, {
+      "type": "string",
+      "name": "Body",
+      "value": "Strong"
+    }, {
+      "type": "string",
+      "name": "Head",
+      "value": "Round"
+    }, {
+      "type": "string",
+      "name": "Tail",
+      "value": "Short"
+    }],
+    "localization": {
+      "da-DK": {
+        "url": "https://location.of/the/danish/metadata.json",
+        "hash": "588d7c14883231cfee522479cc66565fd9a50024603a7b8c99bd7869ca2f0ea3"
+      }
+    }
+  }
+
+The danish localization JSON file could be:
+
+.. code-block:: json
+
+  {
+    "name": "Bibi - Ryan katten",
+    "description": "Ryan katte er ensomme væsner, som rejser rundt i galaxen søgende efter deres forfædre og sande fortid"
+  }
+
+Smart contract limitations
+==========================
+
+A number of limitations are important to be aware of:
+
+- Smart contract function parameters are limited to 1 KiB.
+- Each logged event is limited to 0.5 KiB.
+- The number of logged events is limited to 64 per contract function invocation.
```

Comment:
That is also how I would interprete it.

per contract _function_ invocation.

But it might be better to use _invoke calls_ since we have moved from message passing.
What do you think?